velox/functions/sparksql/Arithmetic.h

```diff
@@ -229,4 +229,13 @@ struct Log1pFunction {
     return true;
   }
 };
+
+template <typename T>
+struct CotFunction {
+  template <typename TInput>
+  FOLLY_ALWAYS_INLINE void call(TInput& result, TInput a) {
+    result = 1 / std::tan(a);
```

Comment:
Does this work? Don't you need to make 1 into a double? 1.0 ?

Reply:
std::tan(a) will return double, so it will be double divide.